rules/PSR4/FileRelocationResolver.php

```diff
@@ -161,8 +161,8 @@ private function replaceRelativeFilePathsWithBeforeAfter(
     ): string {
         // A. first "dir has changed" dummy detection
         $relativeFilePathParts = Strings::split(
-            $oldSmartFileInfo->getRelativeFilePath(),
-            '#' . DIRECTORY_SEPARATOR . '#'
+            $this->normalizeDirectorySeparator($oldSmartFileInfo->getRelativeFilePath()),
+            '#' . preg_quote(DIRECTORY_SEPARATOR, '#') . '#'
```

Comment:
actually this PR includes also https://github.com/rectorphp/rector-src/pull/116.
but since https://github.com/rectorphp/rector-src/pull/116 fixes a separate PR beforehand, I would suggest merging https://github.com/rectorphp/rector-src/pull/116 first